### PR TITLE
[stable/helm] initialize chart in helm repo

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: greg <greg.hill@monax.io>

This PR adds a stable helm chart for [Hoard](https://github.com/monax/hoard), our go based, multi-backend object store. It deterministically encrypts any file sent to it and stores it at the configured location which may be publicly accessible. For example, [Monax](https://monax.io) uses this technology to save legal documentation in a way which can be referenced on our blockchain and easily audited by permissible legal bodies.

cc: @compleatang 

## Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/hoard/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+To uninstall/delete the `my-release` deployment:
+
+```bash
+$ helm delete my-release
+```
+
+## Configuration
+
+The following table lists the configurable parameters of the Burrow chart and its default values.
```
Comment (compleatang) [inline stable/hoard/README.md]: `s/Burrow/Hoard/g`

File: stable/hoard/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Hoard
+
+[Hoard](https://github.com/monax/hoard) is a stateless, deterministically encrypted, content-addressed object store.
```
Comment (compleatang) [inline stable/hoard/README.md]: Maybe add another sentence here about what hoard does.... In terms of utilizing various backends to store files that are encrypted or not. Also mentioning it's semantic encryption framework.

File: stable/hoard/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+version: 0.6.0
+appVersion: 1.1.5
+description: Hoard is a stateless, deterministically encrypted, content-addressed object store
+home: https://github.com/monax/hoard
+icon: https://pbs.twimg.com/profile_images/781959787856687105/76s1CJER_400x400.jpg
+keywords:
+- s3
+- aws
+- gcp
+- envelope encryption
```
Comment (compleatang) [inline stable/hoard/Chart.yaml:11]: - content addressable
- distributed file storage

File: stable/hoard/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+The following table lists the configurable parameters of the Burrow chart and its default values.
+
+| Parameter | Description | Default |
+| --------- | ----------- | ------- |
+| `replicaCount` | number of daemons | `1` |
+| `image.repository` | docker image | `"quay.io/monax/hoard"` |
+| `image.tag` | version | `"1.1.5"` |
+| `image.pullPolicy` | pull policy | `"IfNotPresent"` |
+| `storage.type` | backend object store | `"local"` |
```
Comment (compleatang) [inline stable/hoard/README.md]: let's list the options in the description field

File: stable/hoard/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+| `ingress` | settings for ingress | `{}` |
+| `resources` | pod resources | `{}` |
+| `nodeSelector` | optional settings | `{}` |
+| `tolerations` | optional settings | `[]` |
+| `affinity` | session affinity | `{}` |
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```bash
+$ helm install --name my-release stable/hoard
```
Comment (compleatang) [inline stable/hoard/README.md:68]: add a quick example for s3.

File: stable/hoard/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,86 @@
+apiVersion: apps/v1
```
Comment (compleatang) [inline stable/hoard/templates/deployment.yaml:1]: We should throw if storage.type is "s3" or "gcs" but no credentialsSecret is given.

File: stable/hoard/README.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+| --------- | ----------- | ------- |
+| `replicaCount` | number of daemons | `1` |
+| `image.repository` | docker image | `"quay.io/monax/hoard"` |
+| `image.tag` | version | `"1.1.5"` |
+| `image.pullPolicy` | pull policy | `"IfNotPresent"` |
+| `storage.type` | backend object store | `"local"` |
+| `storage.region` | object store location | `""` |
+| `storage.bucket` | object storage container | `""` |
+| `storage.prefix` | bucket folder | `"hoard"` |
+| `storage.credentialsSecret` | optional secret | `""` |
```
Comment (compleatang) [inline stable/hoard/README.md]: this is required if gcs or s3 are selected.

**Review by compleatang — CHANGES_REQUESTED** [review]

**Comment by compleatang** [comment]
For easy processing by helm/charts team we should rename the PR to `[stable/helm] initialize chart in helm repo`.

Also, it should be noted for reviewers that this chart has been used from the https://github.com/agreements-network/charts repository for about six months by a broad variety of users and as such is fairly well tested. This is why we are PR'ing it into stable rather than incubator. Also it is a pretty straight-forward chart.

**Comment by davidkarlsen** [comment]
Please follow https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md too

**Comment by compleatang** [comment]
@davidkarlsen this chart and its runtime doesn't have any need to talk to the k8s API; as such I'm not sure the RBAC implications. Are you saying that every chart should have RBAC boilerplate even if it doesn't need to communicate to the k8s API? We're happy either way as it's easy enough to add in the boilerplate, but I just want to clarify your comment. Thanks!

**Comment by compleatang** [comment]
ping @unguiculus @davidkarlsen

File: stable/hoard/.helmignore [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+.svn/
+# Common backup files
+*.swp
+*.bak
+*.tmp
+*~
+# Various IDEs
+.project
+.idea/
+*.tmproj
```
Comment (cpanato) [inline stable/hoard/.helmignore:22]: add the `OWNERS` file here as well

File: stable/hoard/templates/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+kind: Deployment
+metadata:
+  name: {{ template "hoard.fullname" . }}
+  labels:
+    app: {{ template "hoard.name" . }}
```
Comment (cpanato) [inline stable/hoard/templates/deployment.yaml]: also please use the new labels, see: https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#metadata

Comment (gregdhill) [reply stable/hoard/templates/deployment.yaml]: done

File: stable/hoard/.helmignore (hunk 8 above)
Comment (gregdhill) [reply stable/hoard/.helmignore:22]: done

File: stable/hoard/templates/deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+      app: {{ template "hoard.name" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "hoard.name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
```
Comment (cpanato) [inline stable/hoard/templates/deployment.yaml]: fix indentation here

File: stable/hoard/templates/deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+{{- if eq .Values.storage.type "local" }}
+          volumeMounts:
+          - mountPath: /data
+            name: data-dir
+{{- end }}
+          resources:
+{{ toYaml .Values.resources | indent 12 }}
+{{- if eq .Values.storage.type "local" }}
+      volumes:
+        - name: data-dir
```
Comment (cpanato) [inline stable/hoard/templates/deployment.yaml]: here as well

File: stable/hoard/templates/ingress.yaml [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
+spec:
+{{- if .Values.ingress.tls }}
+  tls:
+{{- range .Values.ingress.tls }}
+    - hosts:
+{{- range .hosts }}
+        - {{ . }}
```
Comment (cpanato) [inline stable/hoard/templates/ingress.yaml]: fix indentation here

File: stable/hoard/templates/ingress.yaml [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    helm.sh/chart: {{ include "hoard.chart" . }}
+{{- with .Values.ingress.annotations }}
+  annotations:
+{{ toYaml . | indent 4 }}
+{{- end }}
+spec:
+{{- if .Values.ingress.tls }}
+  tls:
+{{- range .Values.ingress.tls }}
+    - hosts:
```
Comment (cpanato) [inline stable/hoard/templates/ingress.yaml]: fix indentation here

File: stable/hoard/templates/ingress.yaml [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+{{- range .hosts }}
+        - {{ . }}
+{{- end }}
+      secretName: {{ .secretName }}
+{{- end }}
+{{- end }}
+  rules:
+{{- range .Values.ingress.hosts }}
+    - host: {{ . }}
```
Comment (cpanato) [inline stable/hoard/templates/ingress.yaml]: fix indentation here

File: stable/hoard/templates/ingress.yaml [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+      http:
+        paths:
+          - path: {{ $ingressPath }}
```
Comment (cpanato) [inline stable/hoard/templates/ingress.yaml]: here as well

File: stable/hoard/templates/ingress.yaml (hunk 12 above)
Comment (cpanato) [reply stable/hoard/templates/ingress.yaml]: there are other places in the templates that need to fix the `- ` indentation

**Comment by gregdhill** [comment]
@cpanato all indentations should be correct now :)

**Comment by gregdhill** [comment]
ping @cpanato

**Review by cpanato — APPROVED** [review]

**Comment by compleatang** [comment]
thanks @cpanato !

**Pull request merged**